### PR TITLE
Sb/dag json

### PR DESCRIPTION
**Pull request by brightly-salty**
Fixes dag-json,
adds quickcheck edid tests
adds documentation

### CODE REVIEW

File: src/dag_cbor.rs [hunk 1]
```diff
@@ -34,10 +35,16 @@ impl TryFrom<u64> for DagCborCodec {
   fn try_from(_: u64) -> core::result::Result<Self, Self::Error> { Ok(Self) }
 }
 
+/// A trait representing the capability to both decode and encode
+/// the type using the dag-cbor codec
 pub trait DagCbor: Encode<DagCborCodec> + Decode<DagCborCodec> {}
 
 impl<T: Encode<DagCborCodec> + Decode<DagCborCodec>> DagCbor for T {}
 
+/// Returns the corresponding dag-json v1 Cid
```
Comment (samuelburnham) [inline src/dag_cbor.rs:44]:
```suggestion
/// Returns the corresponding dag-cbor v1 Cid 
```

File: src/ipld.rs [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+};
 use sp_cid::Cid;
 use sp_std::{
   self,
+  boxed::Box,
   collections::btree_map::BTreeMap,
   vec::Vec,
 };
 
+/// TODO
```
Comment (samuelburnham) [inline src/ipld.rs:14]: I can help with these TODOs after merge if needed.

**Review by samuelburnham — APPROVED** [review]

File: src/ipld.rs (hunk 2 above)
Comment (brightly-salty) [reply src/ipld.rs:14]: Yeah, thanks. I guess I still don't understand references too well yet

**Comment by johnchandlerburnham** [comment]
Looks great

**Pull request merged**